File: src/main/java/net/atos/zac/zoeken/model/ZoekResultaat.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.zoeken.model

import net.atos.zac.shared.model.Resultaat
import java.util.EnumMap

class ZoekResultaat<TYPE>(items: List<TYPE>?, count: Long) :
    Resultaat<TYPE>(items, count) {

    val filters: EnumMap<FilterVeld, List<FilterResultaat>> = EnumMap(
        FilterVeld::class.java
    )

    fun addFilter(facetVeld: FilterVeld, waardes: List<FilterResultaat>) {
        filters[facetVeld] = waardes
    }
}

File: src/main/app/src/app/zoeken/model/zoek-resultaat.ts
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

import { Resultaat } from "../../shared/model/resultaat";
import { GeneratedType } from "../../shared/utils/generated-types";
import { FilterVeld } from "./filter-veld";

/**
 * @deprecated - use the `GeneratedType`
 */
export class ZoekResultaat<TYPE> extends Resultaat<TYPE> {
  filters: Partial<Record<FilterVeld, GeneratedType<"FilterResultaat">[]>> = {};
}
